### DISCUSSION TITLE
Welcome to ColorLCDVape-RE Discussions!

### DISCUSSION THREAD

**Discussion by ginbot86** [ginbot86/ColorLCDVape-RE#1] [Announcements]
(boilerplate introduction text follows):

## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Rea-XEVION** [comment]
Since this post does say to comment below, I might as well be the first.
I usually go by the name Hex on discord servers like the BennVenn and Inside Gadgets chats, and since around 2017 I've been salvaging batteries from disposable vapes of all kinds with the intention to turn empty units into battery banks. Life kept getting in the way so I never found "the perfect" size/shape of 3.7v to 5v boost regulators, nor did I find any small enough multi-battery balancing board to use 2 to 4 batteries in parallel, so eventually I recycled all my batteries and decided to start from nothing with the new screen equipped vapes.

Now I'm more interested in trying to reuse these debuggable / reprogrammable vapes as a sort of "roadkill development board", especially since most of the projects I'm interested in only need like 3 or 4 I/O pins.  The potential of reusing the vape coil power circuit for a 12w / 24w flashlight on my projects is just a bonus.

**Comment by CaptainChristms** [comment]
Hey, I’m the Captain! Hahahaha!

I've been a tinkerer all my life, but I was never able to go very deep with any of my projects because it’s hard to find people to receive guidance from for the things I want to do. For example, the vape screens I recently started pulling eventually led me here, but nowhere else lol. Really hoping I can get some answers and guidance for projects!

**Comment by Estacaco** [comment]
Managed to dump the Firmware off the Geek Bar Skyview,  I couldn't get it to talk with the stuff provided by xbenkozx. I just used the already included ST32F0X.cfg and was able to get 2 way communication going. I did not have a USB C cable to use so I just soldered directly to the DIO, SCLK, 3V and ground and connected it to my STLink.
Would be neat if we could get Micropython on there or something.

**Reply by ginbot86** [reply]
I noticed, thanks for the pull request! The tool by xbenkozx is hard-coded for the Nations N32G031 and its Flash controller addresses. Not all vapes expose the debug interface over the USB-C CC lines either.

**Reply by Estacaco** [reply]
This one does as well I believe, the Via's and the pads i'm using are sitting right under the USB-C port.

**Reply by Estacaco** [reply]
Just another note on this after reading your latest Issues posting. I also had the RST(reset pad hooked up to my STLink, not sure if it was required for the dump but for anyone trying and it not working for them.

**Comment by Estacaco** [comment]
Wife got an Airfuze Vape that's basically a smart watch built into a vape. excited to tear it apart when it's done.

**Reply by ginbot86** [reply]
Neat. I wonder how similar it will be to the Feed Sync I disassembled and mentioned in [issue 9](https://github.com/ginbot86/ColorLCDVape-RE/issues/9)... Maybe even the same JL7012 SoC?

**Reply by Estacaco** [reply]
Yup same JL7012F6. Noticed googling it that its in aome smarth watches as well

**Comment by Infinimitsu** [comment]
I'm getting into robotics and electronics to work on cool projects with my kids.  I've been going down a lot of rabbit holes, lately.  One idea I had for one of the robotics projects we're working on would be using a robot vacuum someone was throwing away as a disc shaped head for the robot we are building as a home assistant.  Originally, I planned on just putting some LEDs behind the "visor" smoked acrylic on the front of the robot sensor bar, and just take all the sensors out.  I've been collecting these vapes, the RAZ 9000 and the RAZ 25000, mainly for the batteries, but recently I decided to look into possibly repurposing the displays for this robotics purpose.  My end goal would be to have several of these displays mounted behind the visor, all able to work together to have the robot be able to emote and have status indicators and stuff, which would definitely be an interesting look, as I haven't really seen anything like it.  I am hoping to learn from this project on how to actually reuse these displays, either with off the shelf boards the displays could be connected to, or reprogramming the board inside the vape to be able to take input from an Arduino or Raspberry Pi and dynamically animate the displays in realtime.  Any help with this is definitely appreciated.  I'm at the end of a 20 year career that had nothing to do with any of this kind of stuff, and I feel like a monkey trying to build a rocketship...

**Reply by Infinimitsu** [reply]
Oh, and my ideal outcome of getting involved here would be to turn the display into just a USB-C connected display.  Possible?

**Reply by ginbot86** [reply]
It's certainly possible, just that there's limitations if one reuses the original vape board. The biggest hurdle to easy image updates is the requirement of an SWD dongle with connections to the USB-C CC pins, and alternative firmware for display control.

**Comment by du-costa** [comment]
![IMG_20241212_084048_577](https://github.com/user-attachments/assets/8dfffb3b-c20f-4c81-889d-3dee100b66ea)
Hi guys, I code like a potato, but I like to tinker. Here's a suggestion. These vapes make perfect meshtastic nodes. Some lora boards have faulty charging circuits so the vape circuit comes to rescue. 
Love to see all this.

**Reply by ginbot86** [reply]
That's actually awesome. I've thought a bit about reusing the housings but have never actually done so.

**Reply by Bobby-H** [reply]
whoooooah, this is awesome! congrats!

**Reply by TheFlemoid** [reply]
That is a fantastic idea!

**Comment by Infinimitsu** [comment]
Considering that at the very least, the batteries can be reused, I went to Goodwill and bought a clear plastic container with a lid and put a "Recycle your vape" label on it and asked my local vape store if they wouldn't mind leaving it on their counter. I came back a week later and had 7 vapes to collect, most were the Raz 25000. I definitely need to figure out a decent project with these or else I'm going to be hoarding nasty used vapes very soon....

**Comment by Bobby-H** [comment]
👋Hello! I'm a tinkerer and DIY nerd who happened by this community by way of a Hackaday article when i guilted over the prospect of these little vaporizers being bound for the dumpster despite the lithium-ion battery, LCD screen, & then-some being contained within. 
My most immediate functional goal is to just continue using the vape (currently struggling to reset the juice meter though i'll post about that separately), though some other ideas i'm fielding are: 
### project ideas
- #### installing the Win95 theme onto the main IC
- -  just trying to follow the tut, pretty much
- #### flashing the main IC to be used in other projects
- - current idea is just using it as the brains of a DIY keyboard? 
im not sure what's possible, though i'd read from the thorough write-ups (thank you so much for those, by the way; i never would have held interest otherwise) that the IC was more capable than it's intended use case lets on. and, if that's successful, then 
- #### reusing the LCD as a little [keyboard display](https://youtu.be/xdAuwUQWaI4?si=ZX4QDMlSZnBiVGzy&t=58)
- - for a bongo-cat, words per minute meter, or what-have-you

But im very grateful to have found this little space for reverse engineering. it's a discipline i've been interested in about as long as i've been coding, i just never found a path of entry into the world until crossing paths with this thing. I wasn't immediately sure of whether or not i'd need to purchase a data probe for any of the above, though i figure i'll purchase that along with a tool to aide with flashing the IC if needed.
